const-oid/src/lib.rs

```diff
@@ -21,50 +25,183 @@
 #![no_std]
 #![doc(
     html_logo_url = "https://raw.githubusercontent.com/RustCrypto/meta/master/logo_small.png",
-    html_root_url = "https://docs.rs/const-oid/0.2.0"
+    html_root_url = "https://docs.rs/const-oid/0.3.0-pre"
 )]
 #![forbid(unsafe_code)]
 #![warn(missing_docs, rust_2018_idioms)]
 
 #[cfg(test)]
 extern crate std;
 
-use core::fmt;
+use core::{convert::TryFrom, fmt, str::FromStr};
+
+/// Maximum number of nodes in an OID.
+///
+/// Note: this is specialized to RustCrypto use cases for now.
+pub const MAX_NODES: usize = 10;
+
+/// Maximum value of the first node in an OID
+const FIRST_NODE_MAX: u32 = 2;
+
+/// Maximum value of the second node in an OID
+const SECOND_NODE_MAX: u32 = 39;
+
+/// Error type
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct Error;
```

Comment:
Implement the `std::error::Error` trait?

Reply:
Yeah, I can do that. So annoying to have to add a `std` feature for that though.